lib/Sema/TypeCheckDecl.cpp

```diff
@@ -8084,6 +8084,8 @@ static Type formExtensionInterfaceType(Type type,
     parentType = unbound->getParent();
     nominal = cast<NominalTypeDecl>(unbound->getDecl());
   } else {
+    if (type->is<ProtocolCompositionType>())
+      type = type->getCanonicalType();
```

Comment:
The very next line casts it to a nominal type, which means this must have failed, right?

Reply:
A protocol composition type where all the terms are the same protocol canonicalizes down to the protocol type itself. A protocol composition with distinct protocol terms is already rejected earlier. Here is an example,

```
protocol P {}
protocol Q {}
typealias A = P
typealias B = P

extension A & B {} // OK!

typealias C = Q

extension A & C {} // error: non-nominal type 'A & C' (aka 'P & Q') cannot be extended
```

Yeah, it's bad.

Reply:
Aw, that's terrible. Do we really have to support this use case?